### PR TITLE
docs: add python quickstart sample

### PR DESCRIPTION
**Pull request by loferris**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-connection/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x ] Ensure the tests and linter pass
- [x ] Code coverage does not decrease (if any source code was changed)
- [ x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/snippets/quickstart.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    for connection in client.list_connections(request=req):
+        print(f"\tConnection {connection.friendly_name} ({connection.name})")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--project_id", type=str)
+    parser.add_argument("--location", default="US", type=str)
+    args = parser.parse_args()
+    main(project_id=args.project_id, location=args.location)
```
Comment (kurtisvg) [inline samples/snippets/quickstart.py:46]: nit: Should this be included in the snippet? generally we favor not including CLIs in new samples

File: samples/snippets/quickstart.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+# [START bigqueryconnection_quickstart]
+import argparse
+
+from google.cloud import bigquery_connection_v1 as bq_connection
+
+
+def main(project_id: str = "your-project-id", location: str = "US") -> None:
+    # Constructs client for interacting with the service
+    client = bq_connection.ConnectionServiceClient()
```
Comment (kurtisvg) [inline samples/snippets/quickstart.py:23]: Can we push the client initialization down into `list_connections`?

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: samples/snippets/quickstart.py (hunk 1 above)
Comment (tswast) [inline samples/snippets/quickstart.py:46]: This was an old pattern. It's difficult to test the CLI arguments, so we omit them in our samples rubric. go/code-snippets-style#running-from-the-command-line

**Review by tswast — COMMENTED** [review]
Looks good. Just one nit.

File: samples/snippets/quickstart.py (hunk 2 above)
Comment (loferris) [reply samples/snippets/quickstart.py]: Addressed in latest commit! I also removed the client fixture from `conftest.py` though I'm guessing we'll need it with future samples!

File: samples/snippets/quickstart.py (hunk 1 above)
Comment (loferris) [reply samples/snippets/quickstart.py]: Removed in latest commit!

Comment (loferris) [reply samples/snippets/quickstart.py]: Awesome, good to know - removed in latest commit!

**Comment by loferris** [comment]
Comments addressed! @tswast @kurtisvg

File: samples/snippets/noxfile.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+# modify the test configurations. The values here should be in sync
+# with `noxfile_config.py`. Users will copy `noxfile_config.py` into
+# their directory and modify it.
+
+TEST_CONFIG = {
+    # You can opt out from the test for specific Python versions.
+    "ignored_versions": [],
+    # Old samples are opted out of enforcing Python type hints
+    # All new samples should feature them
+    "enforce_type_hints": False,
```
Comment (kurtisvg) [inline samples/snippets/noxfile.py:45]: nit: Do we want to flip this to true?

**Review by kurtisvg — APPROVED** [review]

File: samples/snippets/noxfile.py (hunk 3 above)
Comment (tswast) [reply samples/snippets/noxfile.py:45]: Already done via `noxfile_config.py`.

File: samples/snippets/quickstart.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+def main(project_id: str = "your-project-id", location: str = "US") -> None:
+    # Constructs client for interacting with the service
+    list_connections(project_id, location)
+
+
+def list_connections(
+    project_id: str = "your-project-id", location: str = "US",
+) -> None:
+    """Prints details and summary information about connections for a given admin project and location"""
```
Comment (tswast) [inline samples/snippets/quickstart.py:28]: We could simplify even further to keep the developer from having to jump around.

```suggestion
def main(project_id: str = "your-project-id", location: str = "US") -> None:
    """Prints details and summary information about connections for a given admin project and location"""
```

**Review by tswast — APPROVED** [review]
Thanks!

**Pull request merged**